### PR TITLE
Updates to ideally support majority of tpch queries

### PR DESCRIPTION
**Pull request by jacques-n**
- Remove aggregate expressions type from generalized expressions. (only allow aggregate expressions as root expressions for aggregation)
- Update function mapping to support options
- Remove named structs from type unions (should only be used in special places as root, not in arbitrary hierarchy)
- Add project, join, fetch, aggregate, sort, set logical relational operations.
- Introduce key scalar and aggregate functions in functions yaml.
- Remove old extensions docs

### CODE REVIEW

File: extensions/aggregate_functions.yaml [hunk 1]
```diff
@@ -1,18 +1,61 @@
...
+    options:
+      overflow: [SILENT, SATURATE, ERROR]
+    variants:
+      - name: scalar
+        decomposable: MANY
+        parameters:
+          - K: [i8,i16,i32,i64,fp32,fp64]
+        arguments:
+          - type: K?
+        intermediate: "STRUCT<K,i64>"
```
Comment (wesm) [inline extensions/aggregate_functions.yaml:38]: Nit: the accumulator probably needs to be the widest type for the particular type class, how would that be expressed (e.g. int16 -> int64 accumulator)?

**Review by wesm — COMMENTED** [review]
Couple small questions on reading

File: binary/relations.proto [hunk 2]
```diff
@@ -83,17 +88,88 @@ message ReadRel {
...
+    Rel right = 3;
+    Expression expression = 4;
+    Expression post_join_filter = 5;
+
+    enum JoinType {
+        UNKNOWN = 0;
+        INNER = 1;
+        OUTER = 2;
+        LEFT = 3;
+        RIGHT = 4;
```
[the remark this answers is not in the document]
Comment (jacques-n) [reply binary/relations.proto:109]: Yup, #44 .

File: binary/relations.proto [hunk 3]
```diff
@@ -83,17 +88,88 @@ message ReadRel {
...
+message AggregateRel {
+    RelCommon common = 1;
+    Rel input = 2;
+    repeated Grouping groupings = 3;
+    repeated Measure measures = 4;
+    Expression.AggregationPhase phase = 5;
+
+    message Grouping {
+        repeated int32 input_fields = 1;
+    }
```
[the remark this answers is not in the document]
Comment (jacques-n) [reply binary/relations.proto:129]: I'm currently modeling as the latter in general algebra. I think it is likely that people will create custom multi-operators (e.g. filter + agg + proj) that do the post aggregate expressions as well but figured that it is better to keep the basic algebra simple. What do you think?

File: extensions/aggregate_functions.yaml (hunk 1 above)
Comment (jacques-n) [reply extensions/aggregate_functions.yaml:38]: Good point. The intermediate should probably just be STRUCT<i64,i64> for this set of functions. Will post an update.

**Pull request merged**